app/uk/gov/hmrc/nrs/retrieval/connectors/NonrepRetrievalConnector.scala

```diff
@@ -32,67 +32,86 @@ package uk.gov.hmrc.nrs.retrieval.connectors
  * limitations under the License.
  */
 
-import play.api.libs.ws.{WSClient, WSResponse}
-import play.api.{Environment, Logger}
+import play.api.Logger
+import play.api.libs.ws.DefaultBodyWritables.writeableOf_String
 import uk.gov.hmrc.http.HeaderNames.explicitlyIncludedHeaders
-import uk.gov.hmrc.http.{HeaderCarrier, HttpReads, HttpResponse}
+import uk.gov.hmrc.http.client.HttpClientV2
+import uk.gov.hmrc.http.{HeaderCarrier, HttpReads, HttpResponse, StringContextOps}
+import uk.gov.hmrc.nrs.retrieval.config.AppConfig
 import uk.gov.hmrc.nrs.retrieval.config.CoreHttpReads.responseHandler
-import uk.gov.hmrc.nrs.retrieval.config.{AppConfig, WSHttpT}
 
 import javax.inject.{Inject, Singleton}
 import scala.concurrent.{ExecutionContext, Future}
 
 @Singleton
-class NonrepRetrievalConnector @Inject()(val environment: Environment,
-                                         val http: WSHttpT,
-                                         ws: WSClient)
-                                        (implicit val appConfig: AppConfig,
-                                         ec: ExecutionContext) {
+class NonrepRetrievalConnector @Inject() (val httpClientV2: HttpClientV2)(using
+  val appConfig: AppConfig,
+  ec: ExecutionContext
+):
 
   private val logger = Logger(this.getClass)
 
-  /**
-   * import uk.gov.hmrc.http.HttpReads.Implicits._ could be used instead,
-   * but here we have an additional log entry for 404 status
-   */
-  implicit val readRaw: HttpReads[HttpResponse] = responseHandler(_, _ , _)
+  /** import uk.gov.hmrc.http.HttpReads.Implicits._ could be used instead, but here we have an additional log entry for 404 status
+    */
+  given readRaw: HttpReads[HttpResponse] = responseHandler(_, _, _)
 
-  def search(queryParams: Seq[(String, String)])(implicit hc: HeaderCarrier): Future[HttpResponse] = {
+  def search(queryParams: Seq[(String, String)])(using hc: HeaderCarrier): Future[HttpResponse] =
```

Comment:
If `hc` is not used then it can be removed ie:
`  (using HeaderCarrier)`
and other paces that use implicits.
(I find this a little confusing as it doesn't look like a variable anymore)